aws/components/apigateway/setup.ftl

```diff
@@ -303,21 +303,57 @@
             outputs={}
             dependencies=apiId
         /]
+
+        [#-- Throttling Configuration --]
+        [#local openapiIntegrations = 
+            getOccurrenceSettingValue(occurrence, [["apigw"], ["Integrations"]], true)]
+
+        [#local methodSettings = [
+            {
+                "HttpMethod": "*",
+                "ResourcePath": "/*",
+                "LoggingLevel": "INFO",
+                "DataTraceEnabled": true
+            } +
+            attributeIfContent(
+                "ThrottlingBurstLimit", 
+                openapiIntegrations.Throttling.BurstLimit) +
+            attributeIfContent(
+                "ThrottlingRateLimit",
+                openapiIntegrations.Throttling.RateLimit)
+        ]]
+
+        [#-- Integration Patterns (as Regex) into Matching Method Throttling (as explicit paths) --]
+        [#if definitionsObject[core.Id]?? ]
+            [#list definitionsObject[core.Id].paths as path,pathConfig]
+                [#list pathConfig?keys as verb]
+                    [#list openapiIntegrations.Patterns as pattern]
+                        [#if path?matches(pattern.Path)] 
+                            [#local methodSettings = combineEntities(
+                                methodSettings,
+                                [
+                                    {
+                                        "ResourcePath" : path,
+                                        "HttpMethod": pattern.Verb
+                                    } +
+                                    attributeIfContent("ThrottlingBurstLimit", pattern.BurstLimit) +
+                                    attributeIfContent("ThrottlingRateLimit", pattern.RateLimit)
+                                ],
+                                ADD_COMBINE_BEHAVIOUR)]
+                        [/#if]
+                    [/#list]
+                [/#list]
+            [/#list]
+        [/#if]
+
         [@cfResource
             id=stageId
             type="AWS::ApiGateway::Stage"
             properties=
                 {
                     "DeploymentId" : getReference(deployId),
                     "RestApiId" : getReference(apiId),
-                    "MethodSettings": [
-                        {
-                            "HttpMethod": "*",
-                            "ResourcePath": "/*",
-                            "LoggingLevel": "INFO",
-                            "DataTraceEnabled": true
-                        }
-                    ],
+                    "MethodSettings": methodSettings,
```

Comment:
We only want to set the defaults if there is throttling settings defined. so I think we want this to be added via `AttributeIfContent()`

Reply:
Yep certainly should be - fixed up